opennms-install/src/main/java/org/opennms/install/Installer.java

```diff
@@ -247,7 +247,7 @@ public void install(final String[] argv) throws Exception {
                 LOG.info(String.format("* using '%s' as the PostgreSQL schema name for OpenNMS", m_migrator.getSchemaName()));
             }
 
-            m_migrator.setupDatabase(m_update_database, m_do_vacuum, m_do_vacuum, m_update_iplike, m_timescaleDB);
+            m_migrator.setupDatabase(m_update_database, m_do_vacuum, m_do_vacuum, m_update_iplike, m_timescaleDB, true);
```

Comment:
I guess this means user needs to install `pgcrypto` extension otherwise upgrade will not succeed.
Can we fallback to using the default id where `pgcrypto` is not installed ?

Reply:
If installing pgcrypto fails in setupDatabase, the exception is caught and logged and the update continues. The liquibase changeset does the same if pgcrypto is not installed with onFail="CONTINUE".

If the user fixes the issue by either upgrading opennms-admin to superuser or manually installing pgcrypto, they can re-run the install script to apply the update.

Reply:
cool. If you validated fallback scenario, this looks good to me !

Reply:
I did. If the install fails, the system sticks with the default ID, but it can be easily fixed at any point in the future.

Reply:
@christianpape I *think* I set up a transaction correctly, can you double check it?